cmake/onnxruntime_framework.cmake

```diff
@@ -8,12 +8,28 @@ file(GLOB_RECURSE onnxruntime_framework_srcs CONFIGURE_DEPENDS
 )
 
 if (onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
-file(GLOB_RECURSE onnxruntime_training_framework_torch_srcs CONFIGURE_DEPENDS
-    "${ORTTRAINING_SOURCE_DIR}/core/framework/torch/*.h"
-    "${ORTTRAINING_SOURCE_DIR}/core/framework/torch/*.cc"
-)
-
+  file(GLOB_RECURSE onnxruntime_training_framework_torch_srcs CONFIGURE_DEPENDS
+      "${ORTTRAINING_SOURCE_DIR}/core/framework/torch/*.h"
+      "${ORTTRAINING_SOURCE_DIR}/core/framework/torch/*.cc"
+  )
   list(APPEND onnxruntime_framework_srcs ${onnxruntime_training_framework_torch_srcs})
+  if (onnxruntime_ENABLE_TRITON)
+    file(GLOB_RECURSE onnxruntime_training_framework_triton_srcs CONFIGURE_DEPENDS
+      "${ORTTRAINING_SOURCE_DIR}/core/framework/triton/*.h"
+      "${ORTTRAINING_SOURCE_DIR}/core/framework/triton/*.cc"
+    )
+    list(APPEND onnxruntime_framework_srcs ${onnxruntime_training_framework_triton_srcs})
+  endif()
+elseif(onnxruntime_ENABLE_TRITON)
+  file(GLOB_RECURSE onnxruntime_training_framework_triton_srcs CONFIGURE_DEPENDS
```

Comment:
please add a comment here explaining this change - basically when torch interop is enabled dlpack etc is already included but when torch interop is not enabled only triton is enabled then we need to explicitly include these dependencies.